sdk/utils/configuration.py

```diff
@@ -16,24 +18,35 @@ class Configuration:
     precision: str = "normal"
     extra_config: Optional[Dict[str, Any]] = None
 
-    def to_dict(self):
+    @staticmethod
+    def from_dict(
+        conf: Optional[Union[Configuration, Dict]]
+    ) -> Optional[Configuration]:
+        if conf is None:
+            return None
+        if isinstance(conf, Configuration):
+            return conf
+        dt = conf.pop("dt", Configuration.dt)
+        precision = conf.pop("precision", Configuration.precision)
+        return Configuration(dt=dt, precision=precision, extra_config=conf)
```

Comment:
Allows the automatic conversion of various inputs to a `Configuration` object as required to initialize from a API response.

Very happy to receive suggestions on this.

Reply:
Any way we could make this programmatic ? I.e this function doesn't need to be updated if the configuration class attributes get updated. We can list all the fields using the `dataclasses.fields` function

something like
```python
from dataclasses import fields

base_conf = {}
for field in fields(Configuration):
  if field.name != "extra_config" and field.name in conf:
     base_conf[field.name]=conf.pop(field.name)
return Configuration(**base_conf, extra_config=conf
```

Reply:
Exactly the kind of configuration I was looking for. I didn't think about the `fields()`
implemented

Reply:
Great! Thanks for the unit test as well :)